server/src/main/java/org/spine3/server/projection/ProjectionRepository.java

```diff
@@ -44,14 +53,67 @@
 public abstract class ProjectionRepository<I, P extends Projection<I, M>, M extends Message>
         extends EntityRepository<I, P, M> implements EventDispatcher {
 
-    private EntityStorage<I> entityStorage;
+    /**
+     * The enumeration of statuses in which a Projection Repository can be during its lifecycle.
+     */
+    private enum Status {
+        /**
+         * The repository instance has been created.
+         *
+         * <p>In this status the storage is not yet assigned.
+         */
+        CREATED,
+
+        /**
+         * A storage has been assigned to the repository.
+         */
+        STORAGE_ASSIGNED,
+
+        /**
+         * The repository is getting events from EventStore and builds projections.
+         */
+        CATCHING_UP,
+
+        /**
+         * The repository completed the catch-up process.
+         */
+        ONLINE,
+
+        /**
+         * The repository is closed and no longer accept events.
+         */
+        CLOSED
+    }
+
+    /**
+     * The current status of the repository.
+     */
+    private Status status = Status.CREATED;
+
+    /**
+     * Underlying storage for projections.
+     */
+    private EntityStorage<I> entityStorage;  //TODO:2016-05-03:alexander.yevsyukov: Why isn't it named projectionStorage?
```

Comment:
@AlexanderLitus, please pay attention to this TODO.
It should probably be `projectionStorage` of `ProjectionStorage` type.